django/db/models/query_utils.py

```diff
@@ -126,7 +127,7 @@ def check(self, against, using=DEFAULT_DB_ALIAS):
             query.add_annotation(value, name, select=False)
         query.add_annotation(Value(1), "_check")
         # This will raise a FieldError if a field is missing in "against".
-        query.add_q(self)
+        query.add_q(Q(Coalesce(self, True, output_field=BooleanField())))
```

Comment:
@shangxiao thanks for the investigation on Oracle.

Another alternative worth considering is doing something along the lines of (assuming we added the `supports_coalesce_condition` feature flag)

```suggestion
        if connections[using].features.supports_coalesce_condition:
            check = Q(Coalesce(self, True, output_field=BooleanField()))
        else:
            check = self | Q(IsNull(self, True))
        query.add_q(check)
```

I don't have Oracle setup to figure out if this works but it seemed like it should from reading a few resources online.

Reply:
If `Q(IsNull(self, True))` doesn't work then surely `Exact(Case(When(Q(IsNull(self, True)), 1)), 1)` will.

Reply:
I originally tried this too. (`IsNull()` has an `as_oracle()` method on its super that wraps the expression in case)
The problem here is that case always maps a condition that evaluates to `NULL` as false 😔 (whereas we'd like an explicit test of the boolean expression against `NULL`)

Here's what the Q object with `query.add_q(self | Q(IsNull(self, True)))` translates to in Oracle:

```
SELECT 1 AS "_CHECK"  FROM DUAL WHERE (NULL >= 0 OR CASE WHEN NULL >= 0 THEN True ELSE False END IS NULL)
```

If I manually plug this into Oracle (replacing true/false with 1/0):

```
SQL> SELECT 1 AS "_CHECK"  FROM DUAL WHERE (NULL >= 0 OR CASE WHEN NULL >= 0 THEN 1 ELSE 0 END IS NULL);

no rows selected
```

Reply:
(So just clarifying the above case doesn't appear to pass through NULLs)